File: 1752-arithmetic-subarrays/arithmetic-subarrays.js
/**
 * @param {number[]} nums
 * @param {number[]} l
 * @param {number[]} r
 * @return {boolean[]}
 */
var checkArithmeticSubarrays = function(nums, l, r) {
    let n = nums.length;
    const ans = [];

    for(let i=0; i<l.length; i++) {
        let s = new Set();
        let min = 1e9;
        let max = -1e9;
        for(let curr=l[i]; curr<=r[i]; curr++) {
            s.add(nums[curr]);
            min = Math.min(min, nums[curr]);
            max = Math.max(max, nums[curr]);
        }
        let diff = (max - min) / (r[i] - l[i]);
        if(diff % 1 != 0) {
            ans.push(false);
            continue;
        }
        let curr = min;
        let flag = true;
        for(let k=0; k<=r[i]-l[i]; k++) {
            if(!s.has(curr)) {
                flag = false;
                break;
            }
            curr += diff;
        }
        ans.push(flag);
    }

    return ans;
};

File: 1752-arithmetic-subarrays/arithmetic-subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        ans = []

        for i in range(len(l)):
            curr = set()
            mini = int(1e9)
            maxi = int(-1e9)
            for j in range(l[i], r[i]+1):
                curr.add(nums[j])
                mini = min(mini, nums[j])
                maxi = max(maxi, nums[j])
            
            diff = (maxi - mini) / (r[i] - l[i])
            if diff % 1 != 0:
                ans.append(False)
                continue
            
            start = mini
            flag = True
            for i in range(r[i]-l[i]+1):
                if start not in curr:
                    flag = False
                    break
                start += diff

            ans.append(flag)
        
        return ans